crates/dekaf/src/read.rs

```diff
@@ -257,7 +272,11 @@ impl Read {
             //
             // Note that sequence must increment at the same rate
             // as offset for efficient record batch packing.
-            let kafka_offset = next_offset - 1;
+            let kafka_offset = if let Some(rewrite_from) = self.rewrite_offsets_from {
+                rewrite_from + records.len() as i64
```

Comment:
does this need to be incremented prior to exit to reflect the current batch size? Otherwise, won't the next preview batch re-use the same offsets?

Or, do we not allow multiple batches to be read by the same Read instance (I suspect this is the case?).

Reply:
Forgot to answer this, but yeah we [don't re-use](https://github.com/estuary/flow/pull/1701/files#diff-e2fd79eb5f242ee4bf404432c264fcd06099907f5e2ae432416214b83c6afb7bR605) the same `Read` instance when doing data-preview reads